make/jdk/src/classes/build/tools/generatebreakiteratordata/RuleBasedBreakIteratorBuilder.java

```diff
@@ -1273,7 +1273,7 @@ else if (c == '/') {
         for (int i = 0; i < decisionPointList.size(); i++) {
             int rowNum = decisionPointList.elementAt(i).intValue();
             state = tempStateTable.elementAt(rowNum);
-            state[numCategories] |= END_STATE_FLAG;
+            state[numCategories] |= (short) END_STATE_FLAG;
             if (sawEarlyBreak) {
                 state[numCategories] |= LOOKAHEAD_STATE_FLAG;
```

Comment:
Does this need a cast as well? and also other cases, e.g. line 1019: state[numCategories] = DONT_LOOP_FLAG;?

Reply:
No, it doesn't. `LOOKAHEAD_STATE_FLAG` is defined as `0x2000` which is within the range of `short`. OTOH, `END_STATE_FLAG` is `0x8000` thus a lossy conversion.